In [1]:
import matplotlib.pyplot as plt
import scipy.stats as st
import seaborn as sns
import pandas as pd
import numpy as np

## Wichtigste Übungsaufgaben:
* 6.1 
* 6.2
* 6.3(a) - (d)
---
# Aufgabe 6.1
Ein Weinhändler behauptet, dass die von ihm gefüllten Weinflaschen 70 Zentiliter enthalten. Ein skeptischer Konsument vermutet aber, dass der Weinhändler zuwenig Wein abfüllt und will diese Behauptung überprüfen. Deshalb kauft er 12 Weinflaschen
und misst ihren Inhalt. Er findet:<br />
71, 69, 67, 68, 73, 72, 71, 71, 68, 72, 69, 72 (in Zentiliter)<br />
Nehmen Sie zunächst an, dass die Standardabweichung der Abfüllung im Voraus bekannt ist. Sie beträgt σ = 1.5 Zentiliter. Da die Standardabweichung der Messungen bekannt ist, können wir einen z-Test durchführen. Führen Sie den (einseitigen; in welche Richtung?) Test auf dem 5%- Signifikanzniveau durch. Geben Sie die Modellannahmen, H0, HA, den Verwerfungsbereich, den Wert der Teststatistik und das Testergebnis explizit an.<br />
Formulieren Sie in einem Satz die Schlussfolgerung für den kritischen Konsumenten.

In [2]:
wf = pd.Series([71, 69, 67, 68, 73, 72, 71, 71, 68, 72, 69, 72])

$$H_0 = 70zl$$
$$H_A < 70zl$$

Verwerfungsbereich festlegen (einseitig, da der Konsument sagt: zu wenig)

In [5]:
st.norm.ppf(q=[0.05], loc=70, scale=1.5/np.sqrt(wf.size))

array([69.28775749])

Der Verwerfungsbereich K = ($-\infty$, 69.29]

X_n berechnen

In [4]:
wf.mean()

70.25

Der beobachtete Wert liegt ausserhalb des Verwerfungsbereich --> die Nullhypothese wird nicht verworfen.

Schlussfolgerung: Es kann nicht bewiesen werden, dass der Weinhändler zu wenig abfüllt.

---
# Aufgabe 6.2
Ein Weinhändler behauptet, dass die von ihm gefüllten Weinflaschen 70 Zentiliter enthalten. Ein skeptischer Konsument vermutet aber, dass der Weinhändler zuwenig Wein abfüllt und will diese Behauptung überprüfen. Deshalb kauft er 12 Weinflaschen
und misst ihren Inhalt. Er findet:<br />
71, 69, 67, 68, 73, 72, 71, 71, 68, 72, 69, 72 (in Zentiliter)<br />
Die Standardabweichung der Abfüllungen ist nicht bekannt. Man muss sie also aus den gemachten Stichproben schätzen: --> siehe Aufgabenstellung.<br />
Da die Standardabweichung geschätzt wurde und nicht mehr exakt bekannt ist, kann der z-Test nicht durchgeführt werden. Verwenden Sie nun den t-Test auf dem 5 %-Signifikanzniveau. <br />
Geben Sie die Modellannahmen, H0, HA, den Verwerfungsbereich, den Wert der Teststatistik und das Testergebnis explizit an.<br />
Was ändert sich an obigem Test?

In [6]:
wf = pd.Series([71, 69, 67, 68, 73, 72, 71, 71, 68, 72, 69, 72])

$$Varianzschätzung = 1.96^2$$
$$Std = 1.96$$

In [7]:
wf.std()

1.9598237397554634

Festlegen Hypothesen
$$H_0 = 70zl$$
$$H_A < 70zl$$

Teststatistik: t-Verteilung mit Freiheitsgrad 11 (12-1 Flasche)

Signifikantsniveau: $\alpha$ = 5%

Verwerfungsbereich beim T-Test festlegen

In [16]:
st.t.ppf(q=0.05, df=wf.size-1, loc=70, scale=wf.std()/np.sqrt(wf.size))

68.98397388627933

Verwerfungsbereich K = $(-\infty$, 68.98]

Testentscheid

In [17]:
wf.mean()

70.25

liegt ausserhalb des Verwerfungsbereichs --> $H_0$ bleibt.

#### Standardisierung

In [18]:
st.t.ppf(q=0.05, df=wf.size-1)

-1.7958848187036696

Verwerfungsbereich K = $(-\infty$, -1.796]

T-Test gem. Skript

In [21]:
(np.sqrt(wf.size)*(wf.mean()-70))/wf.std()

0.4418894343490791

gleiches Resultat, das Ergebnis liegt nicht im Verwerfungsbereich

---
# Aufgabe 6.3
Unterhalb einer Kläranlage wurden 16 unabhängige Wasserproben aus einem Fluss entnommen und jeweils deren Ammoniumkonzentration Xi (in μg NH_4-N/l) mit einem Messgerät bestimmt. Der Mittelwert der Proben ergab $x_{16}$ = 204.2. Wir wollen nun wissen, ob mit diesem Experiment eine Überschreitung des Grenzwerts von 200 μg NH_4-N/l nachgewiesen werden kann (auf dem 5% Niveau).
### a) Nehmen Sie an, die Standardabweichung der Messungen sei im Voraus aufgrund früherer Studien bekannt. Sie betrage 10 μg NH_4-N/l. Führen Sie unter dieser Annahme einen z-Test durch, um zu prüfen, ob eine Grenzwertüberschreitung nachgewiesen werden kann.
Geben Sie die Modellannahmen, H0, HA, den Verwerfungsbereich, den Wert der Teststatistik und das Testergebnis explizit an.

Modell: $X_i$ ist eine kontinuierliche Mengengrösse; $\sigma_x$ bekannt = 10<br />
Nullhypothese: $H_0 = 200$<br />
Alternativ: $H_A > 200$<br />
Teststatistik: $N(\mu, \frac{\sigma^2_x}{n})$<br />
Signifikantsniveau: $\alpha$ = 5%

In [25]:
st.norm.ppf(q=[0.95], loc=200, scale=10/np.sqrt(16))

array([204.11213407])

Verwerfungsbereich K = [204.11, $\infty$)

Testentscheid: $x_16$ = 204.2 > 204.11 --> liegt im Verwerfungsbereich also wird die Nullhypothese verworfen.

#### Standardisierung

In [28]:
st.norm.ppf(q=0.95)

1.6448536269514722

$$z = \frac{204.2 - 200}{10 / \sqrt(16)}$$

In [29]:
(204.2-200)/(10/np.sqrt(16))

1.6799999999999955

### b) Wie wahrscheinlich ist es, dass man mit 16 unabhängigen Wasserproben eine Grenzwertüberschreitung nachweisen kann, wenn die wahre Ammoniumkonzentration tatsächlich über dem Grenzwert, und zwar bei 205 μg NH_4-N/l liegt?

Aufgabe a) <-- Alles über 204.11 kann verworfen werden.<br />
Nun müssen wir die W'Keit von $P(X_n > 204.11)$ berechnen.<br />
Und zwar mit dem Erwartungswert 205.<br />
Also rechnen wir: $1 - P(X_n <= 204.11)$ da es demselben Resultat entspricht

In [32]:
1 - st.norm.cdf(x=204.11, loc=205, scale=10/np.sqrt(16))

0.6390797174095532

Die W'keit beträgt also ca. 64%

### c) Wie wahrscheinlich ist es, dass man mit 16 unabhängigenWasserproben fälschlicherweise eine Grenzwertüberschreitung nachweist, obwohl die wahre Ammoniumkonzentration bei 200 μg NH-4-N/l liegt und den Grenzwert somit genau einhält?


--> $H_0$ korrekt, aber verworfen. --> Fehler 1. Art

Fehler 1. Art entsprechen dem Signifikantsniveau --> 5%

### d) Nehmen Sie an, dass die Standardabweichung von 10 μg/l aus den 16 Proben geschätzt worden ist. Deshalb ist nun ein t-Test (Nullhypothese μ0 = 200 μg/l) und nicht ein z-Test angebracht. Führen Sie den t-Test durch.

Modell: $X_i$: i.te Ammoniumbestimmung; $\sigma_x$ unbekannt<br />
Nullhypothese: $H_0 = 200$<br />
Alternativ: $H_A != 200$<br />
Teststatistik: t-Verteilung mit 15 Freiheitsgraden; $T = \frac{\sqrt{n}*(X_n-\mu_0)}{\sigma_x}$ = $ T=\frac{\sqrt{16}*(X_n - 200)}{10}$<br/>
Signifikantsniveau: $\alpha$ = 5%<br />
$x_{16} = 204.2$

In [36]:
st.t.ppf(q=0.95, df=15, loc=200, scale=10/np.sqrt(16))

204.38262588923138

Verwerfungsbereich K = [204.38, $\infty$)

Testentscheid: $x_{16}$ liegt nicht im Verwerfungsbereich --> $H_0$ wird beibehalten

#### Standardisierung

In [37]:
st.t.ppf(q=0.95, df=15)

1.7530503556925547

K = $(1.75, \infty)$

$$T = \frac{x_n - \mu_0}{\sigma / \sqrt{n}} == T = \frac{204.2 - 200}{10 / \sqrt{16}}$$

In [38]:
(204.2-200)/(10/np.sqrt(16))

1.6799999999999955